src/Language/PureScript/Sugar/Names/Common.hs

```diff
@@ -24,14 +24,25 @@ warnDuplicateRefs
   -> m ()
 warnDuplicateRefs pos toError refs = do
   let withoutCtors = deleteCtors `map` refs
-      dupeRefs = mapMaybe (refToName pos) $ withoutCtors \\ nub withoutCtors
+      dupeRefs = mapMaybe (refToName pos) $ removeUnique withoutCtors
       dupeCtors = concat $ mapMaybe (extractCtors pos) refs
 
   for_ (dupeRefs ++ dupeCtors) $ \(pos', name) ->
     warnWithPosition pos' . tell . errorMessage $ toError name
 
   where
 
+  -- Removes all unique elements from list
+  -- Example:
+  --  removeUnique [1,2,2,3,3,3,4] == [2,2,3,3,3]
+  removeUnique :: Eq a => Ord a => [a] -> [a]
+  removeUnique = concat . filter twoOrMore . group . sort
+    where
+      twoOrMore :: [a] -> Bool
```

Comment:
I still think this can just be `(\x -> length x > 1)`, but I'll leave that to your judgement.

Reply:
Going with the "guaranteed fast" over the "likely fast" strategy, even if it's a few more lines of code.